pkg/scheduler/partition.go

```diff
@@ -622,11 +612,10 @@ func (pc *PartitionContext) updatePartitionResource(delta *resources.Resource) {
 func (pc *PartitionContext) addNodeToList(node *objects.Node) error {
 	pc.Lock()
 	defer pc.Unlock()
-	if pc.nodes[node.NodeID] != nil {
+	// Node can be added to the system to allow processing of the allocations
+	if pc.nodes.AddNode(node) != nil {
```

Comment:
it's better to read the error message from the returned err:

```
if err := nc.nodes[node.NodeID]; err != nil {
  return fmt.Errorf("failed to add node %s to partition %s, error: %v", nc.Partition, node.NodeID, err)
}
```

Reply:
Agreed. Will update this.